python-shell/src/gafferpy/gaffer.py

```diff
@@ -2405,6 +2557,253 @@ def to_json(self):
+                        el_def = JsonConverter.from_json(
+                            el_def, ElementFilterDefinition)
+                    self.edges.append(el_def)
+            else:
+                for group, el_def in edges.items():
+                    if not isinstance(el_def, ElementFilterDefinition):
+                        el_def = JsonConverter.from_json(
+                            el_def, ElementFilterDefinition)
+                    el_def.group = group
+                    self.edges.append(el_def)
+
+        if global_elements is not None:
+            if not isinstance(global_elements, GlobalElementFilterDefinition):
+                global_elements = JsonConverter.from_json(
+                    global_elements, GlobalElementFilterDefinition)
+            self.global_elements = global_elements
+
+        if global_entities is not None:
+            if not isinstance(global_entities, GlobalElementFilterDefinition):
+                global_entities = JsonConverter.from_json(
+                    global_entities, GlobalElementFilterDefinition)
+            self.global_entities = global_entities
+
+        if global_edges is not None:
+            if not isinstance(global_edges, GlobalElementFilterDefinition):
+                global_edges = JsonConverter.from_json(
+                    global_edges, GlobalElementFilterDefinition)
+            self.global_edges = global_edges
+
+    def to_json(self):
+        operation = super().to_json()
+        if self.input is not None:
+            elements_json = []
+            for element in self.input:
+                elements_json.append(element.to_json())
+            operation['input'] = elements_json
+
+        if self.entities is not None:
+            el_defs = {}
+            for el_def in self.entities:
+                el_defs[el_def.group] = el_def.to_json()
+            operation['entities'] = el_defs
+        if self.edges is not None:
+            el_defs = {}
+            for el_def in self.edges:
+                el_defs[el_def.group] = el_def.to_json()
+            operation['edges'] = el_defs
+
+        if self.global_elements is not None:
+            operation['globalElements'] = self.global_elements.to_json()
+
+        if self.global_entities is not None:
+            operation['globalEntities'] = self.global_entities.to_json()
+
+        if self.global_edges is not None:
+            operation['globalEdges'] = self.global_edges.to_json()
+
+        return operation
+
+
+class Aggregate(Operation):
+    CLASS = 'uk.gov.gchq.gaffer.operation.function.Aggregate'
+
+    def __init__(self,
+                 input=None,
+                 entities=None,
+                 edges=None,
+                 options=None):
+        super().__init__(
+            _class_name=self.CLASS,
+            options=options
+        )
+        self.input = input
+        self.entities = None
+        self.edges = None
+        self.global_elements = None
```

Comment:
The globals can be removed here, they're not required for the aggregate function.